it/src/main/java/org/corfudb/universe/universe/vm/VmUniverse.java

```diff
@@ -0,0 +1,319 @@
+                return;
+            }
+            universeParams
+                    .getGroupParams(groupName, GroupParams.class)
+                    .<VmServerParams>getNodesParams()
+                    .forEach(serverParams -> {
+                        CorfuServerOnVm node = CorfuServerOnVm.builder()
+                                .universeParams(universeParams)
+                                .params(serverParams)
+                                .vm(vms.get(serverParams.getVmName()))
+                                .build();
+
+                        try {
+                            node.destroy();
+                        } catch (NodeException ex) {
+                            log.info("Can't destroy corfu server: {}", groupName, ex);
+                        }
+                    });
+        });
+    }
+
+    /**
+     * Deploy and power on a VM appliance in vSphere.
+     *
+     * @return VirtualMachine instance
+     */
+    private VirtualMachine deployVm(String vmName) throws NodeException {
+        log.info("Deploy VM: {}", vmName);
+
+        VirtualMachine vm;
+        try {
+            // Connect to vSphere server using VIJAVA
+            VmUniverseParams params = universeParams;
+            ServiceInstance si = new ServiceInstance(
+                    new URL(params.getVSphereUrl()),
+                    params.getVSphereUsername(),
+                    params.getVSpherePassword(),
+                    true
+            );
+
+            InventoryNavigator inventoryNavigator = new InventoryNavigator(si.getRootFolder());
+
+            // First check if a VM with this name already exists or not
+            vm = (VirtualMachine) inventoryNavigator.searchManagedEntity(
+                    ManagedEntityType.VIRTUAL_MACHINE.typeName, vmName);
+            if (vm != null) {
+                // If the VM already exists, ensure the power state is 'on' before return the VM
+                if (vm.getSummary().runtime.powerState == VirtualMachinePowerState.poweredOff) {
+                    log.info(vmName + " already exists, but found in 'poweredOff' state, powering it on...");
+                    Task task = vm.powerOnVM_Task((HostSystem) inventoryNavigator.searchManagedEntity(
+                            "HostSystem", universeParams.getVSphereHost()));
+                    task.waitForTask();
+                }
+            } else {
+                // Find the template machine in the inventory
+                VirtualMachine vmTemplate = (VirtualMachine) inventoryNavigator.searchManagedEntity(
+                        ManagedEntityType.VIRTUAL_MACHINE.typeName,
+                        params.getTemplateVMName()
+                );
+
+                log.info("Deploying the VM {} via vSphere {}...", vmName, params.getVSphereUrl());
+
+                // Create customization for cloning process
+                VirtualMachineCloneSpec cloneSpec = createLinuxCustomization(vmName);
```

Comment:
You can create a utility method to do the cloning.

Reply:
Vm specific methods have extracted into ApplianceManager